pluma/io/glia.py

```diff
@@ -0,0 +1,57 @@
+import struct
+import warnings
+import numpy as np
+
+from typing import Union
+
+import pandas as pd
+from pluma.io.path_helper import ComplexPath, ensure_complexpath
+
+
+def load_glia(filename: str,
+              dtypes: list[list[chr]],
+              channel_names: list[list[str]],
+              root: Union[str, ComplexPath] = ''
+              ) -> pd.DataFrame:
+
+    path = ensure_complexpath(root)
+    path.join(filename+'_Frame1.bin')
+
+    df_timestamps = pd.DataFrame(columns=channel_names[0])
+    df_data = pd.DataFrame(columns=channel_names[1])
+
+    try:
+        # unpack timestamps
+        data_bytes = sum([struct.calcsize(f) for f in dtypes[0]])
+        format_string = ''.join(dtypes[0])
+        with path.open('rb') as stream:
+            while True:
+                chunk = stream.read(data_bytes)
+                if not chunk:
+                    break
+
+                unpacked = list(struct.unpack(format_string, chunk))
+                df_timestamps.loc[len(df_timestamps.index)] = unpacked
```

Comment:
Now that all chunks in each file have the same size and data format, it might be possible to rewrite this loop using `np.fromfile` or other vectorized loading function from numpy or pandas packages.

Reply:
Resolved in [d29195a](https://github.com/emotional-cities/pluma-analysis/pull/32/commits/d29195aab3ffd31ca171070402ab565e6930e652)